app/src/main/java/com/humberto/tasky/agenda/data/task/CreateTaskWorker.kt

```diff
@@ -0,0 +1,28 @@
+package com.humberto.tasky.agenda.data.task
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.humberto.tasky.agenda.domain.task.TaskRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+
+@HiltWorker
+class CreateTaskWorker @AssistedInject constructor(
+    @Assisted context: Context,
+    @Assisted params: WorkerParameters,
+    private val taskRepository: TaskRepository
+): CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        val taskId = inputData.getString("TASK_ID") ?: return Result.failure()
+
+        return try {
+            taskRepository.syncPendingCreateTask(taskId = taskId)
+            Result.success()
+        } catch (e: Exception) {
+            Result.retry()
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video Philipp shows the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc